src/napari_micromanager/_gui_objects/_toolbar.py

```diff
@@ -0,0 +1,368 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Tuple, cast
+
+from fonticon_mdi6 import MDI6
+from napari._qt.widgets.qt_viewer_dock_widget import QtViewerDockWidget
```

Comment:
importing from a private module is always risky.  In this case, since you only need it for a type hint, you should _definitely_ put it inside of `if TYPE_CHECKING`

Reply:
even better ... just use `QtWidgets.QDockWidget` for the type hint.  Notice that outside of `_add_dock_widget`, the only part of the dockwidget api that you use is `show` and `raise` ... so from a typing perspective, you're not using the full `QtViewerDockWidget` object api.  The one exception is the line `dock_wdg._close_btn = False`, but that happens inside the scope of `_add_dock_widget`, so the type checker doesn't care